src/glmfit.jl

```diff
@@ -233,10 +233,13 @@ function _fit!(m::AbstractGLM, verbose::Bool, maxIter::Integer, minStepFac::Real
         installbeta!(p, f)
         crit = (devold - dev)/dev
         verbose && println("$i: $dev, $crit")
-        if crit < convTol
+        if crit < convTol || dev == 0
             cvg = true
             break
         end
+        if !isfinite(crit)
+            error("convergence criterion is $crit") # maybe change ConvergenceException to allow for custom error
+        end
```

Comment:
Using a `StatsBase.ConvergenceException` sounds like a good idea here (this is the kind of thing it was made for), but unfortunately that's only set up to capture the number of iterations rather than the convergence criterion, so the error message might not be as informative...